x/mongo/driver/topology/connection.go

```diff
@@ -108,7 +109,21 @@ func (c *connection) connect(ctx context.Context) {
 	}
 	defer close(c.connectDone)
 
+	c.connectContextMutex.Lock()
 	ctx, c.cancelConnectContext = context.WithCancel(ctx)
+	c.connectContextMutex.Unlock()
+
+	defer func() {
+		var cancelFn context.CancelFunc
```

Comment:
nit: Add whitespace similar to that in the `closeConnectContext` function.

Reply:
done